platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationImpl.java

```diff
@@ -95,6 +99,19 @@ public ApplicationImpl(Repository repo) {
             name = mainFeature.getName();
             version = mainFeature.getVersion();
             description = mainFeature.getDescription();
+        } else if (autoInstallFeatures.size() == 0) {
+            // If the app's features file does not contain any auto install features, set
+            // mainFeature to the first feature found in the file.
```

Comment:
It's not necessarily the first defined feature in the features file that gets set as the main feature, it's the feature that equals the parsed name from the 'Repository' object. That feature should be the root feature of the features file.